internal/httpapi/httpapi.go

```diff
@@ -113,33 +129,40 @@ func (s *Server) Serve(ctx context.Context) {
 		}()
 	}
 
-	if s.cfg.EnableIngest {
+	if s.Ingest.enabled {
 		go func() {
-			s.logger.WithFields(logrus.Fields{
-				"port": s.cfg.PortIngest,
-				"host": s.cfg.HostIngest,
-			}).Debug("Ingest API starting listening.")
-
-			router := httprouter.New()
-			router.GET(ingestAPIPathReady, s.handleReady)
-			router.POST(ingestAPIPath, s.handleIngest)
-			err := http.ListenAndServe(fmt.Sprintf("%s:%d", s.cfg.HostIngest, s.cfg.PortIngest), router)
+			err := s.serveIngest()
 			if err != nil {
-				s.logger.WithError(err).Error("unable to start Ingest-API")
-				return
+				log.WithError(err).Error("Ingest server error")
 			}
-			s.logger.Debug("Ingest API stopped.")
 		}()
 	}
 
 	c := http.Client{}
 	var ingestReady, statusReady bool
 	for {
-		if !ingestReady && s.cfg.EnableIngest {
-			ingestReady = s.isGetSuccessful(c, fmt.Sprintf("http://%s:%d%s", s.cfg.HostIngest, s.cfg.PortIngest, ingestAPIPathReady))
+		if !ingestReady && s.Ingest.enabled {
+			scheme := "http://"
+			if s.Ingest.tls.enabled {
+				scheme = "https://"
+				c.Transport = &http.Transport{
+					TLSClientConfig: &tls.Config{
+						InsecureSkipVerify: true,
+					},
+				}
+			}
+
+			if s.Ingest.tls.validateClient {
+				// If client validation is enabled, we cannot probe the /ready path wihtout a valid certificate, which
```

Comment:
Tikismikis comment: small type in `wihtout`

Reply:
Thanks a lot for catching all those 😅 

You might want to set up https://github.com/codespell-project/actions-codespell here